app/controllers/chapters_controller.rb

```diff
@@ -236,12 +241,33 @@ def destroy
 
   private
 
+  # Check whether we should display :new or :edit instead of previewing or
+  # saving the user's changes.
+  def chapter_is_invalid?
+    # make sure at least one of the pseuds is actually owned by this user
+    if @chapter.authors.present? && (@chapter.authors & current_user.pseuds).empty?
```

Comment:
Would it be worthwhile to break this out into its own method, like `user_can_use_pseud` (which is a terrible name)? I don't feel super strongly about it, and it's definitely not worth it if it's a pain in the neck, but checking the user's ownership of a pseud just feels a little out of place in a method called `chapter_is_invalid?`

Reply:
At this point, `@chapter.authors` has been set. If the chapter were to be saved at this point, it would remove the current editor as one of the chapter's creators. So it's sort of invalid in the sense that the chapter shouldn't be saved with its current attributes, even though `@chapter.valid?` would return true.

I could rename the whole function to something else, like `chapter_cannot_be_saved?` or something. Just to differentiate between things that should prevent the chapter from being saved and things that make the chapter invalid.

Reply:
Renaming sounds good!